poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -475,27 +514,19 @@ public void checkAndUpdateIntegrationStatus(FacilitySoftwareConfig powerIQ, Stri
          return assets;
       }
       Map<String, Asset> pduAssetMap = getPDUAssetMap();//get all pdus from flowgate
```

Comment:
you need to modify the getPDUAssetMap() method ,since it is still hardcode to read all data from nlyte , beside that we should not try to load all PDU from our database, instead we should only load PDUs belong to this PowerIQ.
Also we can only read the PDU once. since I guess you will also need the PDU data when sync the PDU information later.